src/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -71,4 +89,95 @@ public enum JsonTokenType : byte
         public bool TryGetInt64Value(out long value) { throw null; }
         public bool TryGetSingleValue(out float value) { throw null; }
     }
+    public ref partial struct Utf8JsonWriter
+    {
+        private object _dummy;
+        public Utf8JsonWriter(System.Buffers.IBufferWriter<byte> bufferWriter, System.Text.Json.JsonWriterState state = default(System.Text.Json.JsonWriterState)) { throw null; }
+        public long BytesCommitted { get { throw null; } }
+        public long BytesWritten { get { throw null; } }
```

Comment:
Are these commonly accessed on the writer?  Wondering when you'd use these rather than getting the CurrentState and using its BytesCommitted/Written.

Reply:
> Are these commonly accessed on the writer?

I can see them being useful if the user is doing synchronous writing and wants to get how much has been written so far. For that pattern, the user never really has to deal with state, so forcing them to call state to get meaningful data that the writer already tracks seems unnecessary.

For example, @KrzysztofCwalina needs it for his scenario: https://github.com/Azure/azure-sdk-for-net-lab/blob/4d8369fe96e99514c108dacb2cd39bc589b0042a/Configuration/Azure.Configuration/ConfigurationSettingParser.cs#L30

With the change from https://github.com/dotnet/corefx/pull/34425#discussion_r245928912, BytesCommitted == BytesWritten when you access them on the CurrentState (since it forces you to flush). If the user wants to know the difference between how much has been written versus committed so far (without flushing). they would need the properties on Utf8JsonWriter.

We could remove **one of** BytesCommitted or BytesWritten from the **CurrentState** though.